include/nbl/system/CArchiveLoaderZip.h

```diff
@@ -127,6 +127,8 @@ class CFileArchiveZip : public IFileArchive
 private:
 	core::smart_refctd_ptr<system::IFile> m_file;
 	bool m_isGZip;
+	core::vector<SZipFileEntry> m_fileInfo;
+	size_t m_readOffset = 0;
```

Comment:
dont have a global `m_readOffset` you wont be able to read the ZIP archive from multiple threads

Also why is the `m_file` you're using not memory mapped?